hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMDBCheckPointServletInodeBasedXfer.java

```diff
@@ -0,0 +1,207 @@
+          Path path = Paths.get(snapshotDir);
+          snapshotDbPaths.add(path);
+        }
+        snapshotDbPaths.add(sstBackupDir);
+        snapshotDbPaths.add(compactionLogDir);
+      }
+
+      boolean shouldContinue = true;
+      for (Path snapshotDbPath : snapshotDbPaths) {
+        if (!shouldContinue) {
+          break;
+        }
+        shouldContinue = getFilesForArchive(copyFiles, hardlinkFiles,
+            sstFilesToExclude, snapshotDbPath, maxTotalSstSize);
+      }
+
+      if (shouldContinue) {
+        shouldContinue = getFilesForArchive(copyFiles, hardlinkFiles,
+            sstFilesToExclude, rdbStore.getDbLocation().toPath(), maxTotalSstSize);
+      }
+
+      writeFilesToArchive(copyFiles, hardlinkFiles, archiveOutputStream,
+          shouldContinue, checkpoint.getCheckpointLocation());
+
+    } catch (Exception e) {
+      LOG.error("got exception writing to archive " + e);
+      throw e;
+    }
+  }
+
+  private OzoneConfiguration getConf() {
+    return ((OzoneManager) getServletContext()
+        .getAttribute(OzoneConsts.OM_CONTEXT_ATTRIBUTE))
+        .getConfiguration();
+  }
+
+  private void writeFilesToArchive(Map<String, Path> copyFiles, Map<String, Set<Path>> hardlinkFiles,
+      ArchiveOutputStream<TarArchiveEntry> archiveOutputStream, boolean completed, Path checkpointLocation)
+      throws IOException {
+    Map<String, Path> filteredCopyFiles = completed ? copyFiles :
+        copyFiles.entrySet().stream().filter(e -> e.getValue().getFileName().toString().toLowerCase().endsWith(".sst"))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+    Path metaDirPath = getMetaDirPath(checkpointLocation);
+    long bytesWritten = 0L;
+    int filesWritten = 0;
+    long lastLoggedTime = Time.monotonicNow();
+
+    // Go through each of the files to be copied and add to archive.
+    for (Map.Entry<String, Path> entry : filteredCopyFiles.entrySet()) {
+      Path path = entry.getValue();
+      // Confirm the data is in the right place.
+      if (!path.toString().startsWith(metaDirPath.toString())) {
+        throw new IOException("tarball file not in metadata dir: "
+            + path + ": " + metaDirPath);
+      }
+
+      File file = path.toFile();
+      if (!file.isDirectory()) {
+        filesWritten++;
+      }
+      LOG.info("Writing file = {} for inode {}", file.getAbsolutePath(), entry.getKey());
+      bytesWritten += includeFile(file, entry.getKey(), archiveOutputStream);
+      // Log progress every 30 seconds
+      if (Time.monotonicNow() - lastLoggedTime >= 30000) {
+        LOG.info("Transferred {} KB, #files {} to checkpoint tarball stream...",
+            bytesWritten / (1024), filesWritten);
+        lastLoggedTime = Time.monotonicNow();
+      }
+    }
+
+    /*
+    if (completed) {
+      // TODO:
+      //  1. take an OM db checkpoint ,
+      //  2. hold a snapshotCache lock,
+      //  3. copy remaining files
```

Comment:
We are also taking a checkpoint of the compaction log directory & SST backup directory and copying remaining files here right? If any compactions completed after snapshots were transferred, that would capture the all the required source files.